image/options.go

```diff
@@ -39,9 +41,41 @@ func WithBuildOptions(options build.ImageBuildOptions) BuildOption {
 type PullOption func(*pullOptions) error
 
 type pullOptions struct {
-	pullClient  ImagePullClient
-	pullOptions image.PullOptions
-	pullHandler func(r io.ReadCloser) error
+	pullClient    ImagePullClient
+	pullOptions   image.PullOptions
+	pullHandler   func(r io.ReadCloser) error
+	credentialsFn func(string) (string, string, error)
+}
+
+// WithCredentialsFn set the function to retrieve credentials for an image to be pulled
+func WithCredentialsFn(credentialsFn func(string) (string, string, error)) PullOption {
```

Comment:
suggestion: we can probably have just one `WithCredentials` option that receives an interface with only one method that returns the username, password and error. We can default to our internal implementation, reading the conf, but we open the gate to anybody else implement their credentials "helpers" Wdyt?

Reply:
You're right. I focused in this changeset. Let's keep it as you suggested: using the function. Thanks for the clarification